### ISSUE TITLE
space-before-function-paren: Generics anonymous functions will report error

### ISSUE THREAD

**Issue by jinasonlin** [typescript-eslint/typescript-eslint#506]
**What code were you trying to parse?**

```ts
const foo = function <T>(t: T): void { console.info(t); };
```

**rule**

```
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "always",
        "named": "never",
        "asyncArrow": "always"
      }
    ]
```

**What did you expect to happen?**
should be consistent with the following
```js
const foo = function (t) { console.info(t); };
```

**What actually happened?**
rule `"anonymous": "always"` will report: Missing space before function parentheses.eslint(space-before-function-paren)

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.7.1-alpha.28` |
| `TypeScript`                | `3.4.5` |
| `ESLint`                    | `5.3.0` |
| `node`                      | `8.x` |

**Comment by armano2** [comment]
`@typescript-eslint/space-before-function-paren` has been added migrated to our package, and it seem that anynymus functions with generic no longer report an issue when paired with always or never

this issue seem to be fixed but, it looks like we are missing/ not reporting errors at all for this rule

@JoshuaKGoldberg we should create documentation for this rule, as its no longer a extension rule (it follow same config) but it doesn't extends base rule anymore

**Comment by JoshuaKGoldberg** [comment]
Hmm, it's a re-implementation? 🤷 documenting the nuances of these base rule extensions/reimplementations/whatnot is pretty low priority IMO. But if somebody wants to file an issue we can accept it.
